### DISCUSSION TITLE
Is shared_secret really needed if the services are communicating using TLS?

### DISCUSSION THREAD

**Discussion by victornoel** [pomerium/pomerium#1432] [General] [answered]
This is maybe a naive question, but while I was trying to deploy the latest version of pomerium, I wondered why there was a `shared_secret` configuration that seems to be used to secure communication between services but at the same time, the services are communicating using TLS.

Isn't `shared_secret` useless in that case or am I (most certainly ;) missing something?

**Comment by desimone** [comment, marked as answer]
Hey @victornoel , great to hear from you.

You are correct that TLS provides transport level integrity and confidentiality when services are communicating directly .  However, `shared_secret` is used to encrypt/sign payloads shared between services via the user's browser (e.g. after OAuth2 callback, or after AuthN on callback) and for encrypting data stored in the databroker. 

Hope that clarifies it!

**Reply by victornoel** [reply]
hehe, indeed, it does clarify, thanks :)

I'm finally trying to update pomerium to latest version but (un)fortunately I'm not back in the ops field, just passing through.
You know my company got bought and we finally have a real ops team here (for now I didn't see any reason to push for pomerium but if I can I will of course, hehe :D)
